tensorflow_datasets/core/features/skip_encoding_feature.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow_datasets as tfds
+from tensorflow_datasets.core.features import feature
+
+
+class SkipEncodingFeature(feature.FeatureConnector):
```

Comment:
This is an interesting approach to skipping encoding. Is it possible to instead check in the base feature whether the data looks encoded? For example, in the `Text` `FeatureConnector`, if the user passes in a list of integers, it could simply skip encoding by checking the input shape/type.

Reply:
My original implementation had this plus reshaping bundled into the base `FeatureConnector`s. Things just got a bit ugly, especially when allowing for potentially dynamic shapes. At one stage it could accept dense arrays or dicts with shape/encoding/encoding_type keys so the BRLE feature could encode directly from RLE data (that's the use case for `shapenet_r2n2` after all - everything else is just trying to optimize for reusability). Replicating that with the RLE feature connector (not used as-yet, but could be in the future?) and there's suddenly 4 classes (RLE/BRLE connector with static/dynamic shapes) that each need to support multiple different input formats.

I'd be worried about inferring the encoding from the shape/data type alone. What if someone accidentally passes in a dense array with `0`s and `1`s rather than `True`s and `False`s? What if they want to pass in `RLE` data and have it encoded as `BRLE`?

Presumably the creator of the feature connector knows the format they'll be feeding the data. At one stage I had the constructor accept an encoder in the constructor... but if they've created the encoder, why not just have them encode the data directly and tell the `FeatureConnector` not to worry?

I guess I don't love the way `FeatureConnector`s couple encoding with decoding. This was my way to optionally decouple those. As it stands I feel composition is preferable to convoluting an otherwise straight-forward `encode` method. If you can suggest a way of dealing with the shape issue more elegantly the complication would be less of an issue, but I still feel there would be a compelling argument for composition.

Happy to be over-ruled :)

Reply:
After some more thinking I've come to agree `SkipEncoding` isn't a great idea. Even if the input is an encoding, it should be checked for consistent lengths. I'll change this to accept either a fully encoded input of the relevant type or dense inputs.

I still feel a separate feature for transforming/possibly saving shape is appropriate...